src/Uno.UI.RemoteControl.VS/EntryPoint.cs

```diff
@@ -301,27 +288,48 @@ private async Task EnsureServerAsync()
 	{
 		_debugAction?.Invoke($"Starting server (tid:{Environment.CurrentManagedThreadId})");
 
-		if (_process is { HasExited: false })
+		var persistedPorts = (await _dte
+			.GetProjectUserSettingsAsync(_asyncPackage, RemoteControlServerPortProperty, ProjectAttribute.Application))
+			.Distinct(StringComparer.OrdinalIgnoreCase)
+			.Select(str => int.TryParse(str, out var p) ? p : -1)
+			.ToArray();
+		var persistedPort = persistedPorts.FirstOrDefault(p => p > 0);
+
+		var port = _devServer?.port ?? persistedPort;
+		// Determine if the port configuration is incorrect:
+		// - Either no ports or multiple ports are persisted (`persistedPorts is { Length: 0 or > 1 }`).
+		// - Or the currently used port (`port`) does not match the persisted port (`persistedPort`).
+		var portMisConfigured = persistedPorts is { Length: 0 or > 1 } || port != persistedPort;
+
+		if (_devServer is { process.HasExited: false })
 		{
-			_debugAction?.Invoke("Server already running");
+			if (portMisConfigured)
+			{
+				_debugAction?.Invoke($"Server already running on port {_devServer?.port}, but port is not configured properly on all projects. Updating it ...");
+
+				// The dev-server is already running, but at least one project is not configured properly
+				// (This can happen when a project is being added to the solution while opened - Reminder: This EnsureServerAsync is invoked each time a project is built)
+				// We make sure to set the current port for **all** projects.
+				await _dte.SetProjectUserSettingsAsync(_asyncPackage, RemoteControlServerPortProperty, port.ToString(CultureInfo.InvariantCulture), ProjectAttribute.Application);
+			}
+			else
+			{
+				_debugAction?.Invoke($"Server already running on port {_devServer?.port}");
+			}
+
 			return;
 		}
 
-		await _processGate.WaitAsync();
+		await _devServerGate.WaitAsync();
 		try
 		{
-
-			if (EnsureTcpPort(ref _remoteControlServerPort))
+			if (EnsureTcpPort(ref port) || portMisConfigured)
 			{
-				// Update the cookie file, so a rebuild will be triggered
-				_remoteControlConfigCookie ??= Path.GetTempFileName();
-				File.WriteAllText(_remoteControlConfigCookie, _remoteControlServerPort.ToString(CultureInfo.InvariantCulture));
-
-				// Push the new port to the project using global properties
-				_ = _globalPropertiesChanged();
+				// The port has changed, or all application projects does not have the same port number (or is not configured), we update port in *all* user files
+				await _dte.SetProjectUserSettingsAsync(_asyncPackage, RemoteControlServerPortProperty, port.ToString(CultureInfo.InvariantCulture), ProjectAttribute.Application);
```

Comment:
It seems that this is not executed, somehow, I don't see the port in the .user file.

Reply:
Updated the project's output type detection, should fix your issue